vispy/visuals/surface_plot.py

```diff
@@ -107,16 +107,24 @@ def set_data(self, x=None, y=None, z=None, colors=None):
                     x = np.arange(self._z.shape[0])
                 else:
                     x = self._x
-            self.__vertices[:, :, 0] = x.reshape(len(x), 1)
+            if x.ndim == 1:
+                self.__vertices[:, :, 0] = x.reshape(len(x), 1)
+
+            if x.ndim == 2:
+                self.__vertices[:, :, 0] = x  # Copy the 2D data into the appropriate slice
             update_mesh = True
 
-        if new_vertices or y is not None:
-            if y is None:
-                if self._y is None:
-                    y = np.arange(self._z.shape[1])
-                else:
-                    y = self._y
-            self.__vertices[:, :, 1] = y.reshape(1, len(y))
+            if new_vertices or y is not None:
+                if y is None:
+                    if self._y is None:
+                        y = np.arange(self._z.shape[1])
+                    else:
+                        y = self._y
```

Comment:
Any idea why github thinks the above 6 lines are new/changed? These look like they are the same as before. Mostly just curious, not concerned.

Reply:
Is there change in indentation level? If so then it is a problem

Reply:
Ah yes, it does seem to have moved over at least one indent level.

Reply:
Adjusted the indentation

Reply:
Adjusted the indentation in my copy of the project